src/socket.ts

```diff
@@ -319,7 +328,13 @@ export class ConstellationSocket extends EventEmitter {
     }
 
     private rebroadcastEvent(name: string) {
-        this.socket.addEventListener(name, evt => this.emit(name, evt));
+        const socket = this.socket;
+        socket.addEventListener(name, evt => {
+            // Sanity check that the socket is current.
+            if (this.socket === socket) {
```

Comment:
Sanity checks should be formed as assertions:

```js
assert(this.socket === socket, 'Attempted to emit an event on an outdated socket');
```

As a side-effect this will prevent `close` events being emitted after `socket.close()` is called, which the caller probably would not expect -- it breaks from the behaviour of the standard WebSocket class.